codegen/src/codegen.rs

```diff
@@ -281,23 +278,24 @@ impl CodeGen {
         namespace: &TokenStream,
         property_name: &str,
         property: &ResolvedSchema,
-    ) -> Result<(TokenStream, Option<TokenStream>)> {
+    ) -> Result<(TokenStream, Vec<TokenStream>)> {
         match &property.ref_key {
             Some(ref_key) => {
                 let tp = ident(&ref_key.name.to_camel_case());
-                Ok((tp, None))
+                Ok((tp, Vec::new()))
             }
             None => {
                 if is_local_enum(property) {
                     let (tp_name, tp) = create_enum(namespace, property_name, property);
-                    Ok((tp_name, Some(tp)))
+                    Ok((tp_name, vec![tp]))
                 } else if is_local_struct(property) {
                     let id = ident(&property_name.to_camel_case());
                     let tp_name = quote! {#namespace::#id};
                     let tps = self.create_struct(doc_file, property_name, property)?;
-                    Ok((tp_name, Some(tps[0].clone())))
```

Comment:
Here is the bug. I was just getting the first type, which left off the mod that was created.